internal/controller/storageclient_controller.go

```diff
@@ -368,6 +371,7 @@ func (r *storageClientReconcile) reconcilePhases() (ctrl.Result, error) {
 
 	r.storageClient.Status.InMaintenanceMode = storageClientResponse.MaintenanceMode
 
+	resourceMapping := gvkToObjectKeys{}
 	for _, kubeResource := range storageClientResponse.KubeResources {
```

Comment:
What is the benefit of sending the resourceMapping into each case? Why can't you just append to the list in a central place above or below the switch case?

Reply:
other options are 
1. unmarshal the object meta twice, at start or end of the switch
2. unmarshal the object meta once and send it to `reconcileResources`

Reply:
done.